File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/AnotherName.cs
using System;
using System.Runtime.Serialization;

//	AnotherName ::= SEQUENCE {
//		type-id    OBJECT IDENTIFIER,
//		value      [0] EXPLICIT ANY DEFINED BY type-id 
//	}

namespace Aladdin.ASN1.ISO.PKIX
{
	[Serializable]
	public class AnotherName : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<ObjectIdentifier>().Factory(), Cast.N, Tag.Any			), 
			new ObjectInfo(    ImplicitCreator				    .Factory  , Cast.E,	Tag.Context(0)  ), 
		}; 
		// конструктор при сериализации
        protected AnotherName(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public AnotherName(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public AnotherName(ObjectIdentifier typeId, IEncodable value) : 
			base(info, typeId, value) {}

		public ObjectIdentifier TypeId	{ get { return (ObjectIdentifier)this[0]; } } 
		public IEncodable		Value	{ get { return                   this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/AnotherName.java
package aladdin.asn1.iso.pkix;
import aladdin.asn1.*; 
import java.io.*;

//	AnotherName ::= SEQUENCE {
//		type-id    OBJECT IDENTIFIER,
//		value      [0] EXPLICIT ANY DEFINED BY type-id 
//	}

public final class AnotherName extends Sequence<IEncodable>
{
    private static final long serialVersionUID = -2004599008123093969L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(ObjectIdentifier.class).factory(), Cast.N, Tag.ANY		  ), 
		new ObjectInfo(    ImplicitCreator				        .factory  , Cast.E, Tag.context(0)), 
	}; 
	// конструктор при раскодировании
	public AnotherName(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public AnotherName(ObjectIdentifier typeId, IEncodable value) 
	{
		super(info, typeId, value); 
	}
	public final ObjectIdentifier   typeId() { return (ObjectIdentifier)get(0); } 
	public final IEncodable         value () { return                   get(1); }
}
